scripts/generateNTPBackgroundImages.js

```diff
@@ -0,0 +1,118 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const path = require('path')
+const mkdirp = require('mkdirp')
+const fs = require('fs-extra')
+const commander = require('commander')
+const util = require('../lib/util')
+const request = require('request')
+
+const jsonSchemaVersion = 1
+
+const createPhotoJsonFile = (path, body) => {
+  fs.writeFileSync(path, body)
+}
+
+const getImageFileNameListFrom = (photoJsonObj) => {
+  let fileList = []
+  if (photoJsonObj.images) {
+    photoJsonObj.images.forEach((wallpaper) => {
+      fileList.push(wallpaper.source)
+    })
+  }
+  return fileList
+}
+
+const isValidSchemaVersion = (version) => {
+  return version === jsonSchemaVersion;
+}
+
+const validatePhotoData = (photoJsonObj) => {
+  // Make sure the data has a schema version so that clients can opt to parse or not
+  if (!isValidSchemaVersion(photoJsonObj.schemaVersion)) {
+    console.log(`Invalid schema version ${photoJsonObj.schemaVersion}`)
+    return false
+  }
+
+  let isValid = true
+  if (photoJsonObj.images) {
+    photoJsonObj.images.forEach((image) => {
+      if (!image.name || !image.source || !image.author ||
+          !image.link || !image.originalUrl || !image.license) {
+        console.log(`Doesn't have sufficient properties`)
+        console.log(image)
+        isValid = false
+      }
+    })
+  }
+  return isValid
+}
+
+const prepareAssets = (jsonFileUrl, targetResourceDir) => {
+  return new Promise(function (resolve, reject) {
+    let jsonFileBody = '{}'
+
+    // Download and parse jsonFileUrl.
+    request(jsonFileUrl, async function (error, response, body) {
+      if (error) {
+        console.error(`Error from ${jsonFileUrl}:`, error)
+        return reject(error)
+      }
+      if (response && response.statusCode === 200) {
+        jsonFileBody = body
```

Comment:
Might be more clear to flip this - and if response is NOT 200, reject with a specific error (maybe include responseText which may have the details). Then, the parsing below will be guaranteed to have a body 😄

Reply:
Fixed :)